### ISSUE TITLE
Possibility of check but no download?

### ISSUE THREAD

**Issue by WinTDoan** [vslavik/winsparkle#253]
I have a software product where some of our customers have very tight restrictions on installation of software (IT departments need to vet all updates). And the users are often in completely separate divisions (maybe even different buildings) from the IT staff. I would like the software to be able to check for updates and let the _end user_ know that a newer version is available and to let them do the usual choice of "Skip this version" and "Remind me later" so they aren't routinely nudged by the software, but to somehow prevent the "Install Update" choice from operating.

I added a 

		win_sparkle_set_user_run_installer_callback(OnCantUpdate);

handler which I can use to prevent it from actually running the update, but that's only after it downloads it. I would prefer to interrupt before even the download, but there doesn't seem to be an earlier notification that let's me stop the remaining processing.

Note that 95% of our users don't have such an issue (but the ones that are are our biggest customers).

**Comment by vslavik** [comment]
> I would like the software to be able to check for updates and let the end user know that a newer version is available and to let them do the usual choice of "Skip this version" and "Remind me later" so they aren't routinely nudged by the software

This seems wrong to me - you propose to harass the user in a way *they can't do anything about*. If an app shows some window, such as "new version available", it needs to be actionable.

I'd expect this kind of deployment to instead prevent automatic update checks, which you can do with registry entry (or the API).

You can also use `win_sparkle_set_did_find_update_callback()` to show "update available" notification in your UI in non-intrusive way. And I'm always open to PRs making the API more flexible (in ways that are universally useful).

**Comment by WinTDoan** [comment]
These are entities like Central Banks which have very tight security on their computers. The users chafe under it, but have to live with it. I did implement a win_sparkle_set_did_find_update_callback() but it doesn't offer a return code for indicating that it was handled by the callback. The only callback which seems to have that feature is

win_sparkle_set_user_run_installer_callback

which is too late in the process (I can prevent it from actually running the installer, but not from downloading the installer). Ideally, a callback which notifies when the user clicks the "Install Update" which offers a 1-0 return similarly for 1=already handled, 0=default Sparkle handler. That way the user can see the release notes and then choose the skip or postpone options so it won't bother them.

BTW, it doesn't look like win_sparkle_set_update_dismissed_callback works on the update UI dialog. I put a handler in on that, and it doesn't respond to either the "Install Update" or to click on the close box. It *does* however, seem to work on the alert box that the software is already up-to-date. win_sparkle_set_update_postponed_callback and win_sparkle_set_update_skipped_callback seem to work fine.
